astropy/time/tests/test_basic.py

```diff
@@ -911,6 +911,22 @@ def test_explicit_longdouble(self):
         assert np.allclose(t2.value, mjd_long,
                            rtol=0., atol=np.finfo(float).eps)
 
+    @pytest.mark.skipif(np.finfo(np.longdouble).eps >= np.finfo(float).eps,
+                        reason="long double is the same as float")
+    def test_explicit_longdouble_one_val(self):
+        """Ensure either val1 or val2 being longdouble is possible.
+
+        Regression test for issue gh-10033.
+        """
+        i = 54321
+        f = max(2.**(-np.finfo(np.longdouble).nmant) * 65536,
```

Comment:
Can you explain what this is doing?

Reply:
This is better understood in the context of other tests above:
https://github.com/astropy/astropy/blob/eb9445eba6fa399ccfde2314a5e7006becb7d1b3/astropy/time/tests/test_basic.py#L895-L898